src/components/checkbox/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,79 @@
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import classNames from 'classnames';
+import { useId } from 'react';
+import { Icon, IconType } from '../../icon';
+
+export type CheckboxState = RadixCheckbox.CheckedState;
+
+export interface ICheckboxProps extends Omit<RadixCheckbox.CheckboxProps, 'asChild'> {
+    /**
+     * Label of the checkbox.
+     */
+    label: string;
+    /**
+     * Position of the label.
+     * @default right
+     */
+    labelPosition?: 'right' | 'left';
+}
+
+export const Checkbox: React.FC<ICheckboxProps> = (props) => {
+    const { label, labelPosition = 'right', id, className, ...otherProps } = props;
+
+    // Generate random id if id property is not set
+    const randomId = useId();
+    const processedId = id ?? randomId;
+
+    return (
+        <div
+            className={classNames(
+                'group/root flex items-center',
+                { 'flex-row': labelPosition === 'right' },
+                { 'flex-row-reverse': labelPosition === 'left' },
+                className,
+            )}
+        >
+            <RadixCheckbox.Root
+                id={processedId}
+                className={classNames(
+                    'group/checkbox peer rounded-[4px]',
+                    'focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset',
```

Comment:
We use this so much that we really should consider making it it's own tw class.

Reply:
Yep makes sense to investigate 👌🏻